File: sdk/scala-sdk-protobuf/src/main/scala/kalix/scalasdk/view/ViewContext.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.scalasdk.view

import kalix.scalasdk.Context

/** Context for views. */
trait ViewContext extends Context {

  /**
   * The id of the view that this context is for.
   *
   * @return
   *   The view id.
   */
  def viewId: String
}

File: sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/view/ViewContext.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.javasdk.view;

import kalix.javasdk.Context;

/** Context for views. */
public interface ViewContext extends Context {
  /**
   * The id of the view that this context is for.
   *
   * @return The view id.
   */
  String viewId();
}
